src/main/java/juja/microservices/gamification/service/AchievementService.java

```diff
@@ -217,53 +219,49 @@ private void addThanksKeeperScheduled() {
         } else {
-            achievementIds = getIdsCreatedAchievement(achievements);
+            achievementIds = achievements.stream()
+                    .map(Achievement::getId)
+                    .collect(Collectors.toList());
+             logger.info("Returned already created THANKS KEEPER achievements in current week {}", achievementIds);
         }
         return achievementIds;
     }
 
     private List<Achievement> getThanksKeeperAchievements() {
-        logger.debug("Send request to repository: get all thanks_keepers achievements for current week");
+        logger.debug("Send request to repository of created THANKS KEEPER achievements for current week");
         List<Achievement> achievements = achievementRepository.getAllThanksKeepersAchievementsCurrentWeek();
-        logger.debug("Received list of achievements, size = {}", achievements.size());
+        logger.debug("Received list of THANKS KEEPER achievements, size = {}", achievements.size());
         return achievements;
     }
 
     private List<String> createThanksKeeperAchievements() {
-        List<String> result = new ArrayList<>();
         logger.debug("Request to keepers repository: get all active keepers");
         List<KeeperDTO> keepers = keeperService.getKeepers();
         logger.debug("Received list of active keepers, size = {} ", keepers.size());
+        List<String> result = new ArrayList<>();
         if (keepers.isEmpty()) {
             logger.debug("No any active keepers received from user service");
         } else {
-            logger.debug("Sending 'Thanks Keeper' achievements to repository");
-            for (KeeperDTO keeper : keepers) {
-                List<String> directions = keeper.getDirections();
-                directions.forEach(direction -> {
-                    String description = String.format(THANKS_DESCRIPTION, direction);
-                    Achievement achievement = new Achievement(
-                            systemFrom, keeper.getUuid(), KEEPER_THANKS, description, AchievementType.THANKS_KEEPER);
-                    result.add(achievementRepository.addAchievement(achievement));
-                });
-            }
-            logger.debug("Received ids from repository: {}", result);
+            result = keepers.stream()
+                    .map(this::getNewThanksKeeperAchievements)
+                    .flatMap(Collection::stream)
+                    .map(achievementRepository::addAchievement)
+                    .collect(Collectors.toList());
         }
         return result;
     }
 
-    private List<String> getIdsCreatedAchievement(List<Achievement> achievements) {
-        List<String> result = getIds(achievements);
-        logger.info("Returned already created 'Thanks Keeper' achievements in current week {}", result.toString());
-
-        return result;
+    private List<Achievement> getNewThanksKeeperAchievements(KeeperDTO keeper) {
```

Comment:
Maybe add some debug logs?
This method invoked in stream, but logs are only before stream in createThanksKeeperAchievements.
We don't know what happens in this method and method 'getAchievement'.
Very much logic is not covered by logs, I think.

Reply:
I changed method's name  to "prepareThanksKeeperAchievements" and added debug logger in the last commit

Reply:
Sorry. I missed this.
Maybe complete this line with keeperDTO?
logger.debug("Preparing THANKS KEEPER achievements for send to repository"); 
In case of some exception we wil know what keeper contain problem.

Reply:
Yes, of course! I added it. Now it looks like this:
`Preparing THANKS KEEPER achievements for send to repository. KeeperDTO(uuid=0002A, directions=[Codenjoy])`